dashboard/pages/community.py

```diff
@@ -0,0 +1,945 @@
+    Args:
+        df_releases (DataFrame): A DataFrame containing filtered data related to new releases
+    Returns:
+        dict: A Python dictionary containing a DataFrame with table formatted data and a table title
+    """
+    df_merged = format_data_for_table(df_releases)
+
+    df_merged = df_merged.sort_values(
+        by=["Number of Reviews"], ascending=False)
+    df_merged = format_columns(df_merged)
+
+    df_merged = df_merged.reset_index(drop=True)
+
+    return {"table_data": df_merged, "title": "Top Recommended Games by Sentiment"}
+
+
+def tokenize_review_text(df_releases: DataFrame) -> DataFrame:
+    """
+    Normalize and tokenize review text for each review and add this data to a new column
+
+    Args:
+        df_releases (DataFrame): A DataFrame containing filtered data related to new releases
+
+    Returns:
+        DataFrame: A DataFrame containing filtered data related to new releases with a new
+        column for tokenized review_text
+    """
+    df_releases.dropna(subset=["review_text"], inplace=True)
+    df_releases["keywords"] = df_releases["review_text"].str.lower()
+    df_releases["keywords"] = df_releases["keywords"].apply(word_tokenize)
+
+    return df_releases
+
+
+def get_wordnet_tags(tokens: list) -> list:
+    """
+    Get the pos tags for a set of tokens, and return the tokens in a way the 
+    lemmatizer can interpret
+
+    Args:
+        tokens (list): A list containing tokens extracted from review text
+
+    Returns:
+        list: A list of tags which the lemmatizer can interpret
+    """
+    tag_map = defaultdict(lambda: "n")
+    tag_map['J'] = "a"
+    tag_map['V'] = "v"
+    tag_map['R'] = "r"
+
+    tagged_tokens = pos_tag(tokens)
+
+    tagged_tokens = [(token[0], tag_map[token[1][0]])
+                     for token in tagged_tokens]
+
+    return tagged_tokens
+
+
+def lemmatize_tokens(df_releases: DataFrame) -> DataFrame:
+    """
+    Lemmatize tokens text for each tokenized review
+
+    Args:
+        df_releases (DataFrame): A DataFrame containing filtered data related to new releases
+
+    Returns:
+        DataFrame: A DataFrame containing filtered data related to new releases with a lemmatization
+        applied to the tokenized review_text column
+    """
+    lemma = WordNetLemmatizer()
+
+    df_releases["keywords"] = df_releases["keywords"].apply(get_wordnet_tags)
+    df_releases["keywords"] = df_releases["keywords"].apply(
+        lambda tokens: [lemma.lemmatize(word=token[0], pos=token[1]) for token in tokens])
+
+    return df_releases
+
+
+def get_filtered_tokens(tokens):
+    """
+    Filter out punctuation, stop words, and very short words 
+
+    Args:
+        tokens (list): A list containing lemmatized tokens 
+
+    Returns:
+        list: A list containing filtered tokens
+    """
+    stops = stopwords.words("english")
+
+    stops.extend(["n't", "play", "get", "still", "game", "like"])
```

Comment:
why are we not including n't words?

Reply:
Dan did not seem to like them for some reason. I'll remove them from the list.